infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsnext/brokerphases/WaitBrokerResult.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2012-2018 Red Hat, Inc.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.workspace.infrastructure.kubernetes.wsnext.brokerphases;
+
+import com.google.common.annotations.Beta;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.api.workspace.server.wsnext.model.ChePlugin;
+
+/**
+ * Wait until Che plugin broker future finishes and returns resulting workspace tooling or error.
+ * Also calls next {@link BrokerPhase}.
+ *
+ * <p>This API is in <b>Beta</b> and is subject to changes or removal.
+ *
+ * @author Oleksandr Garagatyi
+ */
+@Beta
+public class WaitBrokerResult implements BrokerPhase {
+
+  private final CompletableFuture<List<ChePlugin>> toolingFuture;
+
+  public WaitBrokerResult(CompletableFuture<List<ChePlugin>> toolingFuture) {
+
+    this.toolingFuture = toolingFuture;
+  }
+
+  @Override
+  public List<ChePlugin> execute() throws InfrastructureException {
+    try {
+      return toolingFuture.get(3, TimeUnit.MINUTES);
```

Comment:
should we make this value configurable?

Reply:
Yes, but it doesn't block us and can be done in a separate PR